### PR TITLE
Origin to sample

### PR DESCRIPTION
**Pull request by RasmusBurge-CG**
### Fixed

**Steps to consider while deploying**
- Configuration changes:
- Documentation updates:
- Inform users by email:

### Review:
- [ ] Code approved by
- [ ] Tests executed on stage by:  (Document the test done with screen shots and description.)
- [ ] "Merge and deploy" approved by

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg_lims/EPPs/udf/copy/orgwell_to_sample.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+from email.encoders import encode_quopri
```
Comment (mayabrandi) [inline cg_lims/EPPs/udf/copy/orgwell_to_sample.py:1]: Please rename the file to original_well_to_sample or something

File: cg_lims/EPPs/udf/copy/orgwell_to_sample.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+from genologics.entities import Process, Artifact
+from genologics.lims import Lims
+
+from cg_lims import options
+from cg_lims.exceptions import LimsError, InvalidValueError, MissingUDFsError
+from cg_lims.get.artifacts import get_artifacts
+
+LOG = logging.getLogger(__name__)
+
+def org_well_to_sample(
```
Comment (mayabrandi) [inline cg_lims/EPPs/udf/copy/orgwell_to_sample.py:16]:
```suggestion
def original_well_to_sample(
```

File: cg_lims/EPPs/udf/copy/orgwell_to_sample.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                    sample.udf['Original Container'] = artifact.location[0].name
+                    sample.put()
+                    passed += 1
+                except:
+                    LOG.error(
+                    f"Error: Sample {artifact.id} missing udf location. Can therefor not assign values."
+                )
+                    failed += 1    
+        else:
+            if artifact.parent_process:
```
Comment (mayabrandi) [inline cg_lims/EPPs/udf/copy/orgwell_to_sample.py:44]: Place the following if statements before the assignment of the sample udfs. You can then delete the redundant line 31 and the code will be flattened.

**Review by mayabrandi — COMMENTED** [review]
Nice nice! I had a few comments

**Comment by RasmusBurge-CG** [comment]
Hi @mayabrandi! Made the changes we talked about! Can you review it again?:)

File: cg_lims/EPPs/udf/copy/original_well_to_sample.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            continue
+
+        elif len(artifact.samples)!=1:
+            LOG.error(
+                f"Error: more than one sample per artifact for {artifact.id}. Assumes a 1-1 relation between sample and artifact."
+            )
+            failed += 1
+            continue
+
+        for sample in artifact.samples:
```
Comment (mayabrandi) [inline cg_lims/EPPs/udf/copy/original_well_to_sample.py:45]: you can skip this loop. There is only going to be one sample according to line 38

```suggestion
sample = artifact.samples[0]
```

**Comment by mayabrandi** [comment]
Nice Rasmus! I had one last comment!

File: cg_lims/EPPs/udf/copy/original_well_to_sample.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
+from email.encoders import encode_quopri
```
Comment (Karl-Svard) [inline cg_lims/EPPs/udf/copy/original_well_to_sample.py:1]: Is this import used?

File: cg_lims/EPPs/udf/copy/original_well_to_sample.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+    Arguments:
+        artifacts: list of artifacts to copy from"""
+
+    failed = 0
+    passed = 0
+    for artifact in artifacts:        
+        if artifact.parent_process:
+            LOG.error(
+            f"Error: Not the first step for sample {artifact.id}. Can therefor not get the original container."
```
Comment (Karl-Svard) [inline cg_lims/EPPs/udf/copy/original_well_to_sample.py:33]:
```suggestion
                f"Error: Not the first step for sample {artifact.id}. Can therefor not get the original container."
```

File: cg_lims/EPPs/udf/copy/original_well_to_sample.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+sample = artifact.samples[0]
+            try:
+                sample.udf['Original Well'] = artifact.location[1]
+                sample.udf['Original Container'] = artifact.location[0].name
+                sample.put()
+                passed += 1
+            except:
+                LOG.error(
+                f"Error: Sample {artifact.id} missing udf location. Can therefor not assign values."
```
Comment (Karl-Svard) [inline cg_lims/EPPs/udf/copy/original_well_to_sample.py:53]:
```suggestion
                    f"Error: Sample {artifact.id} missing udf location. Can therefor not assign values."
```

File: cg_lims/EPPs/udf/copy/original_well_to_sample.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+                failed += 1
+
+    if failed:
+        raise InvalidValueError(
+            message=f"Failed to set 'Original Well' and 'Original Container' for {failed} artifacts. Udfs were set on {passed} samples."
+        )
+
+@click.command()
+@options.input(
+    help="Use this flag if you want copy udfs from input artifacts. Defaulte is output artifacts."
```
Comment (Karl-Svard) [inline cg_lims/EPPs/udf/copy/original_well_to_sample.py:64]:
```suggestion
    help="Use this flag if you want copy udfs from input artifacts. Default is output artifacts."
```

**Review by Karl-Svard — APPROVED** [review]
Nice work! I just have a few comments.

File: cg_lims/EPPs/udf/copy/original_well_to_sample.py (hunk 5 above)
Comment (RasmusBurge-CG) [reply cg_lims/EPPs/udf/copy/original_well_to_sample.py:1]: No, probably not :)

Comment (RasmusBurge-CG) [reply cg_lims/EPPs/udf/copy/original_well_to_sample.py:1]:
```suggestion
```

**Pull request merged**